Testing/VelaptorTests/Content/Caching/SoundCacheTests.cs

```diff
@@ -0,0 +1,367 @@
+
+            // Act, & Assert
+            AssertExtensions.ThrowsWithMessage<ArgumentNullException>(() =>
+            {
+                cache.GetItem(filePath);
+            }, "The parameter must not be null or empty. (Parameter 'soundFilePath')");
+        }
+
+        [Fact]
+        public void GetItem_WithUnsupportedFileType_ThrowsException()
+        {
+            // Arrange
+            const string dirPath = @"C:\my-sounds\";
+            const string soundName = "test-sound";
+            const string invalidExtension = ".txt";
+            var soundFilePath = $"{dirPath}{soundName}{invalidExtension}";
+            var exceptionMsg = $"Sound file type '{invalidExtension}' is not supported.";
+            exceptionMsg += $"\nSupported file types are '{OggFileExtension}' and '{Mp3FileExtension}'.";
+
+            this.mockPath.Setup(m => m.GetExtension(soundFilePath)).Returns(invalidExtension);
+
+            var cache = CreateCache();
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<LoadSoundException>(() =>
+            {
+                cache.GetItem(soundFilePath);
+            }, exceptionMsg);
+        }
+
+        [Fact]
+        public void GetItem_WhenOggFileDoesNotExist_ThrowsException()
+        {
+            // Arrange
+            this.mockPath.Setup(m => m.GetExtension(this.oggSoundFilePath)).Returns(OggFileExtension);
+            this.mockFile.Setup(m => m.Exists(this.oggSoundFilePath)).Returns(false);
+            var cache = CreateCache();
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<FileNotFoundException>(() =>
+            {
+                cache.GetItem(this.oggSoundFilePath);
+            }, $"The '{OggFileExtension}' sound file does not exist.");
+        }
+
+        [Fact]
+        public void GetItem_WhenMp3FileDoesNotExist_ThrowsException()
+        {
+            // Arrange
+            this.mockPath.Setup(m => m.GetExtension(this.mp3SoundFilePath)).Returns(Mp3FileExtension);
+            this.mockFile.Setup(m => m.Exists(this.mp3SoundFilePath)).Returns(false);
+            var cache = CreateCache();
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<FileNotFoundException>(() =>
+            {
+                cache.GetItem(this.mp3SoundFilePath);
+            }, $"The '{Mp3FileExtension}' sound file does not exist.");
+        }
+
+        [Fact]
+        public void GetItem_WhenGettingSound_ReturnsSound()
+        {
+            // Arrange
+            var mockSound = new Mock<ISound>();
+            this.mockSoundFactory.Setup(m => m.Create(this.oggSoundFilePath))
+                .Returns(mockSound.Object);
+
+            var cache = CreateCache();
+
+            // Act
+            var sound = cache.GetItem(this.oggSoundFilePath);
+
+            // Assert
+            Assert.NotNull(sound);
+        }
+
+        [Fact]
+        public void Unload_WhenSoundToUnloadExists_RemovesAndDisposesOfTexture()
```

Comment:
- [ ] Refactor unit test method name
- [ ]  Also do a search for the word **Texture** or any variants and refactor to sound where applicable.

```suggestion
        public void Unload_WhenSoundToUnloadExists_RemovesAndDisposesOfSound()
```